src/main/java/ca/uhn/fhir/jpa/starter/ExportInterceptor.java

```diff
@@ -0,0 +1,38 @@
+package ca.uhn.fhir.jpa.starter;
+import ca.uhn.fhir.rest.server.interceptor.*;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.api.server.ResponseDetails;
+import ca.uhn.fhir.jpa.starter.ExportData;
+import java.util.*;
+import java.io.*;
+
+public class ExportInterceptor extends InterceptorAdapter {
+
+   /**
+    * Override the incomingRequestPreProcessed method, which is called
+    * for each incoming request before any processing is done
+    */
+   @Override
+   public boolean incomingRequestPreProcessed(HttpServletRequest theRequest, HttpServletResponse theResponse) {
+     String endPoint = theRequest.getRequestURL().substring(theRequest.getRequestURL().lastIndexOf("/")+1);
+     if (endPoint.equals("$export")) {
```

Comment:
Add check to make sure this is a GET request
```suggestion
     if (endPoint.equals("$export") && theRequest.getMethod().equals("GET")) {
```